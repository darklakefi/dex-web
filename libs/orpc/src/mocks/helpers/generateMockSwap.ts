import {
  randBetweenDate,
  randBoolean,
  randCurrencySymbol,
  randNumber,
  randUuid,
  seed,
} from "@ngneat/falso";
import type { Swap } from "../../schemas";

export function generateMockSwap(randomSeed: string) {
  seed(randomSeed);

  return {
    buyAmount: randNumber({ max: 1000000, min: 1 }),
    buyBalance: randNumber({ max: 1000000, min: 1 }),
    buyToken: {
      address: randUuid(),
      decimals: randNumber({ max: 1000000, min: 1 }),
      symbol: randCurrencySymbol(),
    },
    estimatedFeesUsd: randNumber({ max: 1000000, min: 1 }),
    exchangeRate: randNumber({ max: 1000000, min: 1 }),
    mevProtectionEnabled: randBoolean(),
    priceImpactPercentage: randNumber({ max: 100, min: 1 }),
    sellAmount: randNumber({ max: 1000000, min: 1 }),
    sellBalance: randNumber({ max: 1000000, min: 1 }),
    sellToken: {
      address: randUuid(),
      decimals: randNumber({ max: 1000000, min: 1 }),
      symbol: randCurrencySymbol(),
    },
    slippageTolerancePercentage: randNumber({ max: 100, min: 1 }),
    swapProgressStep: randNumber({ max: 3, min: 1 }),
    swapStatus: "pending",
    swapType: "swap",
    transactionSignature: randUuid(),
    updatedAt: randBetweenDate({
      from: new Date("2025-01-01"),
      to: new Date("2025-06-24"),
    }).toISOString(),
    userAddress: randUuid(),
  } satisfies Swap;
}
