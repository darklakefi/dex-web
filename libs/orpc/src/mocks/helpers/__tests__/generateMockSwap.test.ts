import { generateMockSwap } from "../generateMockSwap";

describe("generateMockSwap", () => {
  it("should generate a swap", () => {
    const response = generateMockSwap();

    expect(response).toMatchSnapshot();
    expect(response.buyAmount).toBeDefined();
    expect(response.buyBalance).toBeDefined();
    expect(response.buyToken).toBeDefined();
    expect(response.estimatedFeesUsd).toBeDefined();
    expect(response.exchangeRate).toBeDefined();
    expect(response.mevProtectionEnabled).toBeDefined();
    expect(response.priceImpactPercentage).toBeDefined();
    expect(response.sellAmount).toBeDefined();
  });
});
