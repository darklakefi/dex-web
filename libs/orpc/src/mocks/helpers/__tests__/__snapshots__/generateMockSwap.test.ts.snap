// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateMockSwap > should generate a swap 1`] = `
{
  "buyAmount": 461381,
  "buyBalance": 142191,
  "buyToken": {
    "address": "a6b9508e-bc09-49c4-87ad-61b6972a8e01",
    "symbol": "NT$",
    "value": "552933",
  },
  "estimatedFeesUsd": 490620,
  "exchangeRate": 755785,
  "mevProtectionEnabled": false,
  "priceImpactPercentage": 41,
  "sellAmount": 571763,
  "sellBalance": 206541,
  "sellToken": {
    "address": "8e74165e-af13-475c-8bf8-2931018191f1",
    "symbol": "B/.",
    "value": "906256",
  },
  "slippageTolerancePercentage": 37,
  "swapProgressStep": 2,
  "swapStatus": "pending",
  "swapType": "swap",
  "transactionSignature": "3bad0ded-7e04-4974-aaa7-9b84152a3d19",
  "updatedAt": "2025-06-24T11:10:14.100Z",
  "userAddress": "06e9095b-f3a2-4241-8599-0dd2469bba8b",
}
`;
